### ISSUE TITLE
Need a better handler for the auth IPs

### ISSUE THREAD

**Issue by jesuismaxime** [selvinortiz/craft-plugin-patrol#24]
It would be better if the plugin uses the database to store the IPs or at least merge the list from the `patrol.php` file with one from the `project.yaml`. The fact that only one of them is working is an issue when working with multiple environment. I juste wiped out my production IPs list because my local environment doesn’t have all the production auth IPs.

This will also allow changes in production when the Craft setting `allowAdminChanges` is net to `false`. For now it display a system error because the plugin isn’t able to push changes to the yaml file.

**Comment by engram-design** [comment]
Agreed. Not _quite_ sure what the thinking was for [re-saving plugin settings](https://github.com/verbb/patrol/blob/2d91980eacf6bdb9e6a90907b926296dddc1d12d/src/services/Service.php#L45) but that won't work at all on environments with `allowAdminChanges = false`.

**Comment by jesuismaxime** [comment]
Any update regarding this? Really need to have Patrol activated on staging/production environments where admin changes are not allowed.

**Comment by engram-design** [comment]
No update yet, sorry - but it'll take a little bit to get that sorted, and backwards-compatible.

Planning this to only be available for Craft 4 though, if that factors in on your end.

**Comment by bartrylant** [comment]
A bit late to the party bit I just bumped into this on Craft 5. The token adding an ip is a great function, but I have admin changes disabled on staging and live servers, so it just throws a server error.
